browser-extension/src/addTournesolRecommendations.js

```diff
@@ -225,11 +226,17 @@ function process() {
   isPageLoaded = true;
   if (videos.length > 0) {
     displayRecommendations();
+  } else if (!areRecommendationsLoaded) {
+    // If the content script is loaded on a non-root URL the recommendations
+    // are not loaded. So if the user then goes to the root URL and the content
+    // script is not reloaded, we need to load the recommendations.
+    loadRecommandations();
   }
 }
 
 function handleResponse({ data: videosReponse }) {
   areRecommandationsLoading = false;
+  areRecommendationsLoaded = true;
```

Comment:
Optionally, a nice refactor here may be to keep track of `recommendationsRequestStatus` first equal to `"idle"`, `"loading"`, then `"loaded"`. So that we don't have the risk to forget to modify both `areRecommandationsLoading` and `areRecommendationsLoaded`